src/repository/index.ts

```diff
@@ -154,15 +159,15 @@ export default class Repository extends EventEmitter implements EventEmitter {
 
   private handleFlagsFromStream(event: { data: string }) {
     try {
-      const data: ClientFeaturesResponse = JSON.parse(event.data);
-      this.save(data, true);
+      const data: ClientFeaturesDelta = JSON.parse(event.data);
+      this.saveDelta(data);
     } catch (err) {
       this.emit(UnleashEvents.Error, err);
     }
   }
 
   timedFetch(interval: number) {
-    if (interval > 0 && !this.eventSource) {
+    if (interval > 0 && this.mode.type === 'polling') {
```

Comment:
we added explicit mode in the constructor params since we need more than eventSource (mode + format)